src/Routing/Duplex.purs

```diff
@@ -328,6 +334,44 @@ prop sym (RouteDuplex f g) (RouteDuplex x y) =
 
 infix 2 prop as :=
 
+-- | Combined with `vcase`, builds a Variant where the order of parsing and
+-- | printing matters. As in the example below, the later `vcase`s take priority
+-- | when parsing/printing:
+-- |
+-- | ```purescript
+-- | userRoutes =
+-- |   variant
+-- |     # vcase (SProxy :: _ "list") (pure unit)
+-- |     # vcase (SProxy :: _ "edit") (str segment)
+-- |     # vcase (SProxy :: _ "new") (path "new" $ pure unit)
+-- | ```
+variant :: forall r. RouteDuplex r (Variant ())
+variant = RouteDuplex mempty (Chomp \_ -> Fail EndOfPath)
```

Comment:
Not sure if `Chomp \_ -> Fail EndOfPath` is the best way to have the codec fail here.